### PR TITLE
Add code for creating a dataset using path on datastore

### PR DESCRIPTION
**Pull request by fuhuifang**
# PR into Azure/azureml-cheatsheets

## Checklist

I have:

- [ X] read and followed the [contributing guidelines](https://azure.github.io/azureml-cheatsheets/docs/misc/contributing)

## Changes

- Add a code example for creating a Dataset given a relative path on datastore
- Add a placeholder for creating a Dataset from local files

### CODE REVIEW

**Comment by fuhuifang** [comment]
From https://docs.microsoft.com/en-us/azure/machine-learning/how-to-create-register-datasets:

Upload files from a local directory and create a FileDataset in a single method with the public preview method, upload_directory(). This method is an experimental preview feature, and may change at any time.

File: website/docs/cheatsheets/python/v1/data.md [hunk 1]
```diff
@@ -251,6 +251,44 @@ config.run_config.data_references[data_ref.data_reference_name] = data_ref.to_co
...
+```
+
+### From a datastore
+The code snippet below shows how to create a Dataset of an entire folder given the relative path `local/test/` on `datastore`. Note that this also applies to creating a Dataset of a single file, which could be done using a relative path like `local/test/data.tsv`.
+```python
+import os
+from azureml.core import Dataset
+
+# create input dataset
+dataset_name = os.environ.get('ADL_DATA_SET', 'folder_data')
```
Comment (aminsaied) [inline website/docs/cheatsheets/python/v1/data.md:269]: Nit: For the sake of simplicity, can we avoid reading from env vars and just set name = "<dataset_name>" or something generic?

File: website/docs/cheatsheets/python/v1/data.md [hunk 2]
```diff
@@ -251,6 +251,44 @@ config.run_config.data_references[data_ref.data_reference_name] = data_ref.to_co
...
+folder_data = Dataset.File.from_files(path=(datastore, os.environ.get('ADL_DATA_RELATIVE_PATH', 'local/test/')))
```
Comment (aminsaied) [inline website/docs/cheatsheets/python/v1/data.md:270]: As above, let's just use a generic sounding path here

File: website/docs/cheatsheets/python/v1/data.md [hunk 3]
```diff
@@ -251,6 +251,44 @@ config.run_config.data_references[data_ref.data_reference_name] = data_ref.to_co
...
+dataset = folder_data.register(workspace=ws, name=dataset_name)
+```
+
+### From outputs using `OutputFileDatasetConfig`
+```python
+from azureml.core import ScriptRunConfig
+from azureml.data import OutputFileDatasetConfig
+
+output_data = OutputFileDatasetConfig(
+    destination=(datastore, "test_path/{run-id}"),
```
Comment (aminsaied) [inline website/docs/cheatsheets/python/v1/data.md:280]: Same again, lets use a generic path. Note: the { } assume you are using an f string f"", which this example doesn't. I'd suggest

destination=(datastore, "path/to/output")

File: website/docs/cheatsheets/python/v1/data.md [hunk 4]
```diff
@@ -251,6 +251,44 @@ config.run_config.data_references[data_ref.data_reference_name] = data_ref.to_co
...
+    name="test_name",
+)
+
+config = ScriptRunConfig(
+        source_directory=".",
+        command=["python", "run.py", "--output_dir", output_data.as_mount()],
```
Comment (aminsaied) [inline website/docs/cheatsheets/python/v1/data.md:286]: Nit: I think command is less well known than script and arguments, I'd suggest doing

...
script="run.py",
arguments=["--output_dir", output_data.as_mount()],
...

**Review by aminsaied — CHANGES_REQUESTED** [review]

**Review by aminsaied — APPROVED** [review]

**Pull request merged**